### PR TITLE
Add module "90nvdimm" for NVDIMM support 

### PR DESCRIPTION
**Pull request by mwilck**
This has been on the ML for some time with no feedback. Posting it here now.

### CODE REVIEW

**Comment by mwilck** [comment]
RPM build errors:
        Installed (but unpackaged) file(s) found:
        /usr/lib/dracut/modules.d/90nvdimm/module-setup.sh

... am I expected to fix the Fedora spec file?

**Comment by mwilck** [comment]
TEST: root filesystem on LVM PV on a isw dmraid   [FAILED] 
    make: *** [Makefile:5: TEST-14-IMSM] Error 2

I doubt that my patches were causing this error.

**Comment by haraldh** [comment]
> ```
> RPM build errors:
>     Installed (but unpackaged) file(s) found:
>     /usr/lib/dracut/modules.d/90nvdimm/module-setup.sh
> ```
> 
> ... am I expected to fix the Fedora spec file?

I can do that later on, if you don't want to.

**Comment by haraldh** [comment]
> ```
> TEST: root filesystem on LVM PV on a isw dmraid   [FAILED] 
> make: *** [Makefile:5: TEST-14-IMSM] Error 2
> ```
> 
> I doubt that my patches were causing this error.

yeah, too bad the CI is still flaky.

File: modules.d/90nvdimm/module-setup.sh [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+    # These modules call "nvdimm_bus_register()".
+    local _provider_dirs='=drivers/nvdimm =drivers/acpi =arch/powerpc'
+
+    #instmods() will take care of hostonly
+    dracut_instmods -o -s nvdimm_bus_register $_provider_dirs
+}
+
+# called by dracut
+install() {
+    inst /usr/bin/ndctl
```
Comment (haraldh) [inline modules.d/90nvdimm/module-setup.sh]: Is this for debugging purposes? What needs this in the initramfs?

Comment (haraldh) [reply modules.d/90nvdimm/module-setup.sh]: Maybe better to use:
```
inst_multiple -o ndctl
```
and have it optional and without the path.

Comment (mwilck) [reply modules.d/90nvdimm/module-setup.sh]: > Is this for debugging purposes? What needs this in the initramfs?

This is needed for encrypted NVDIMMs.

    $ cat /etc/modprobe.d/nvdimm-security.conf
    install libnvdimm /usr/bin/ndctl load-keys ; /sbin/modprobe --ignore-install libnvdimm $CMDLINE_OPTS

This is subobtimal; "install" in modprobe.d is more or less deprecated, and this "install" statement is only needed for users of encrypted NVDIMMs which probably represent a tiny minority of the already low number of NVDIMM users. OTOH, without this, booting from encrypted NVDIMM devices simply fails.

Comment (haraldh) [reply modules.d/90nvdimm/module-setup.sh]: Ok, can you change it to ```inst_multiple -o ndctl``` nevertheless?

**Comment by mwilck** [comment]
I've pushed an update to https://github.com/mwilck/dracut/tree/nvdimm. But I can't push to this autogenerated branch. Should I supersede this PR with another one from my tree, or post a v2 to the ML?

**Comment by haraldh** [comment]
> I've pushed an update to https://github.com/mwilck/dracut/tree/nvdimm. But I can't push to this autogenerated branch. Should I supersede this PR with another one from my tree, or post a v2 to the ML?

ah, ok.. supersede this PR with another one from your tree

**Comment by mwilck** [comment]
Closing in favor of #749.

**Pull request closed without merging**